client/src/language/languageClient.ts

```diff
@@ -23,13 +23,25 @@ import {
   TransportKind,
   type ServerOptions
 } from 'vscode-languageclient/node'
-
-const getEmbeddedLanguageDocInfos = async (client: LanguageClient, uriString: string, position: Position): Promise<{ uri: string, lineOffset: number }> => {
-  return await client.sendRequest('custom/getEmbeddedLanguageDocInfos', { uriString, position })
+import { RequestMethod, type RequestParams, type RequestResult } from '../lib/src/types/requests'
+import { NotificationMethod, type NotificationParams } from '../lib/src/types/notifications'
+
+const getEmbeddedLanguageDocInfos = async (
+  client: LanguageClient,
+  uriString: string,
+  position: Position
+): RequestResult['EmbeddedLanguageDocInfos'] => {
+  const params: RequestParams['EmbeddedLanguageDocInfos'] = { uriString, position }
```

Comment:
Is it possible to use the dot notation to reference the property? You won't need to change the value here in case you change the value of the property.

If it is too much trouble, maybe it is easier to define one type for each param/result.